// @bit/bluekeel.feed.featured-row/dist/components/FeaturedRow/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _helperz = require("../../utils/helperz");

var _featured = _interopRequireDefault(require("../../data/featured.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var FeaturedRow = function FeaturedRow() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  var ArticleList = function ArticleList() {
    var componentList = [];

    _featured.default.map(function (article, idx) {
      if (idx === 0 || idx > 5) return null;
      var currentObj = {
        title: article.title,
        categories: article.categories,
        link: article.link,
        content: article.contentSnippet
      };
      componentList.push(currentObj);
      return null;
    });

    return componentList.map(function (article, i) {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "featured-article-list-item",
        onClick: function onClick(e) {
          return handleLinkClick(article.link, e);
        },
        key: "featured-article-list-item-".concat(i)
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "article-title title-hover"
      }, article.title), /*#__PURE__*/_react.default.createElement("div", {
        className: "keyword-link-container bottom-border"
      }, article.categories.map(function (category, j) {
        return /*#__PURE__*/_react.default.createElement("div", {
          className: "keyword-link",
          key: "category-item-".concat(j)
        }, /*#__PURE__*/_react.default.createElement("a", {
          className: "category-item",
          href: article.link
        }, /*#__PURE__*/_react.default.createElement("span", null, category)));
      })));
    });
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "feed-row featured-row-container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "featured-article-container",
    onClick: function onClick(e) {
      return handleLinkClick(_featured.default[0].link, e);
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "featured-article-title title-hover"
  }, _featured.default[0].title), /*#__PURE__*/_react.default.createElement("img", {
    className: "featured-article-img",
    src: (0, _helperz.getSrcFromHtml)(_featured.default[0].content),
    alt: "featured-article-img"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "featured-article-list"
  }, /*#__PURE__*/_react.default.createElement(ArticleList, null)));
};

var _default = FeaturedRow;
exports.default = _default;
