// @bit/bluekeel.feed.popular-row/dist/components/PopularRow/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _helperz = require("../../utils/helperz");

var _popular = _interopRequireDefault(require("../../data/popular.json"));

var _Card = _interopRequireDefault(require("@material-ui/core/Card"));

var _CardActionArea = _interopRequireDefault(require("@material-ui/core/CardActionArea"));

var _CardContent = _interopRequireDefault(require("@material-ui/core/CardContent"));

var _CardMedia = _interopRequireDefault(require("@material-ui/core/CardMedia"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LatestRow = function LatestRow() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  var CardDeck = function CardDeck() {
    var componentList = [];

    _popular.default.map(function (article, idx) {
      if (idx >= 9) return null;
      var currentObj = {
        title: article.title,
        categories: article.categories,
        link: article.link,
        content: article.contentSnippet,
        html: article.content
      };
      componentList.push(currentObj);
      return null;
    });

    return componentList.map(function (article, i) {
      return /*#__PURE__*/_react.default.createElement(_Card.default, {
        key: "article-card_".concat(i),
        className: "article-card"
      }, /*#__PURE__*/_react.default.createElement(_CardActionArea.default, {
        onClick: function onClick(e) {
          return handleLinkClick(article.link, e);
        }
      }, /*#__PURE__*/_react.default.createElement(_CardContent.default, {
        className: "card-body"
      }, /*#__PURE__*/_react.default.createElement("h2", {
        className: "card-title row-column-item-title"
      }, article.title), /*#__PURE__*/_react.default.createElement(_CardMedia.default, {
        component: "img",
        src: (0, _helperz.getSrcFromHtml)(article.html, true, 'lg')
      }), /*#__PURE__*/_react.default.createElement("p", {
        className: "card-text"
      }, (0, _helperz.trimSnippet)(article.content)))));
    });
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "feed-row row__container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bottom-border popular-title row-title"
  }, "Popular"), /*#__PURE__*/_react.default.createElement("div", {
    className: "card-deck"
  }, /*#__PURE__*/_react.default.createElement(CardDeck, null)));
};

var _default = LatestRow;
exports.default = _default;
