// @bit/bluekeel.feed.latest-row/dist/components/LatestRow/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _helperz = require("../../utils/helperz");

var _latest = _interopRequireDefault(require("../../data/latest.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LatestRow = function LatestRow() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  var ArticleList = function ArticleList() {
    var componentList = [];

    _latest.default.map(function (article, idx) {
      if (idx >= 4) return null;
      var currentObj = {
        title: article.title,
        categories: article.categories,
        link: article.link,
        content: article.contentSnippet,
        html: article.content
      };
      componentList.push(currentObj);
      return null;
    });

    return componentList.map(function (article, i) {
      return /*#__PURE__*/_react.default.createElement("div", {
        key: "latest-article-".concat(i),
        className: "latest-article-row",
        onClick: function onClick(e) {
          return handleLinkClick(article.link, e);
        }
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "latest-item-title-column"
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "article-title"
      }, article.title), /*#__PURE__*/_react.default.createElement("div", {
        className: "keyword-link-container tags-left"
      }, article.categories.map(function (category, j) {
        return /*#__PURE__*/_react.default.createElement("div", {
          className: "keyword-link",
          key: "category-item-".concat(j)
        }, /*#__PURE__*/_react.default.createElement("a", {
          className: "category-item",
          href: "https://unitedstatescredit.blog/2018/11/06/earn-100s-by-opening-up-an-online-checking-account-3/"
        }, /*#__PURE__*/_react.default.createElement("span", null, category)));
      }))), /*#__PURE__*/_react.default.createElement("div", {
        className: "text-column"
      }, (0, _helperz.trimSnippet)(article.content)), /*#__PURE__*/_react.default.createElement("div", {
        className: "image-column"
      }, /*#__PURE__*/_react.default.createElement("img", {
        className: "latest-image",
        src: (0, _helperz.getSrcFromHtml)(article.html, true, 'lg'),
        alt: "list-".concat(i)
      })));
    });
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "feed-row row__container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "bottom-border row-title"
  }, "The Latest"), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-content-column"
  }, /*#__PURE__*/_react.default.createElement(ArticleList, null)));
};

var _default = LatestRow;
exports.default = _default;
