// @bit/bluekeel.feed.info-row/dist/components/InfoRow/Reviews.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _helperz = require("../../utils/helperz");

var _reviews = _interopRequireDefault(require("../../data/reviews.json"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Reviews = function Reviews() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: (0, _helperz.getSrcFromHtml)(_reviews.default[2].content, true, 'sm'),
    className: "row-column-item-image",
    alt: "img"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-content"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-title title-hover",
    onClick: function onClick(e) {
      return handleLinkClick(_reviews.default[2].link, e);
    }
  }, _reviews.default[2].title), /*#__PURE__*/_react.default.createElement("div", {
    className: "keyword-link-container small-column-categories"
  }, _reviews.default[2].categories.map(function (category, j) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "keyword-link",
      key: "category-item-".concat(j)
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: "category-item",
      href: _reviews.default[2].link
    }, /*#__PURE__*/_react.default.createElement("span", null, category)));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: (0, _helperz.getSrcFromHtml)(_reviews.default[3].content, true, 'sm'),
    className: "row-column-item-image",
    alt: "img"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-content"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-title title-hover",
    onClick: function onClick(e) {
      return handleLinkClick(_reviews.default[3].link, e);
    }
  }, _reviews.default[3].title), /*#__PURE__*/_react.default.createElement("div", {
    className: "keyword-link-container small-column-categories"
  }, _reviews.default[3].categories.map(function (category, j) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "keyword-link",
      key: "category-item-".concat(j)
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: "category-item",
      href: _reviews.default[3].link
    }, /*#__PURE__*/_react.default.createElement("span", null, category)));
  })))));
};

var _default = Reviews;
exports.default = _default;
