// @bit/bluekeel.feed.info-row/dist/components/InfoRow/Tips.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; if (typeof Symbol === "function" && typeof Symbol.iterator === "symbol") { _typeof = function _typeof(obj) { return typeof obj; }; } else { _typeof = function _typeof(obj) { return obj && typeof Symbol === "function" && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }; } return _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _helperz = require("../../utils/helperz");

var data = _interopRequireWildcard(require("../../data/tips.json"));

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var Tips = function Tips() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  return /*#__PURE__*/_react.default.createElement(_react.default.Fragment, null, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: (0, _helperz.getSrcFromHtml)(data[2].content, true, 'sm'),
    className: "row-column-item-image",
    alt: "img"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-content"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-title title-hover",
    onClick: function onClick(e) {
      return handleLinkClick(data[2].link, e);
    }
  }, data[2].title), /*#__PURE__*/_react.default.createElement("div", {
    className: "keyword-link-container small-column-categories"
  }, data[2].categories.map(function (category, j) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "keyword-link",
      key: "category-item-".concat(j)
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: "category-item",
      href: data[2].link
    }, /*#__PURE__*/_react.default.createElement("span", null, category)));
  })))), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item"
  }, /*#__PURE__*/_react.default.createElement("img", {
    src: (0, _helperz.getSrcFromHtml)(data[3].content, true, 'sm'),
    className: "row-column-item-image",
    alt: "img"
  }), /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-content"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-column-item-title title-hover",
    onClick: function onClick(e) {
      return handleLinkClick(data[3].link, e);
    }
  }, data[3].title), /*#__PURE__*/_react.default.createElement("div", {
    className: "keyword-link-container small-column-categories"
  }, data[3].categories.map(function (category, j) {
    return /*#__PURE__*/_react.default.createElement("div", {
      className: "keyword-link",
      key: "category-item-".concat(j)
    }, /*#__PURE__*/_react.default.createElement("a", {
      className: "category-item",
      href: data[3].link
    }, /*#__PURE__*/_react.default.createElement("span", null, category)));
  })))));
};

var _default = Tips;
exports.default = _default;
