// @bit/bluekeel.feed.editorial-row/dist/components/EditorialRow/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _react = _interopRequireDefault(require("react"));

var _featured = _interopRequireDefault(require("../../data/featured.json"));

var _search_pad = _interopRequireDefault(require("../../assets/search_pad.jpg"));

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var EditorialRow = function EditorialRow() {
  var handleLinkClick = function handleLinkClick(linkout, e) {
    e.preventDefault();
    window.open(linkout, '_blank');
  };

  var handleVisualSearch = function handleVisualSearch(e) {
    e.preventDefault();
    window.scrollTo(0, 0);
  };

  var buildArticleList = function buildArticleList() {
    var componentList = [];

    _featured.default.map(function (article, idx) {
      if (idx <= 5 || idx >= 10) return null;
      var currentObj = {
        title: article.title,
        categories: article.categories,
        link: article.link,
        content: article.contentSnippet
      };
      componentList.push(currentObj);
      return null;
    });

    return componentList.length > 0 && componentList.map(function (article, i) {
      return /*#__PURE__*/_react.default.createElement("div", {
        className: "featured-article-list-item",
        onClick: function onClick(e) {
          return handleLinkClick(article.link, e);
        },
        key: "featured-article-list-item-".concat(i)
      }, /*#__PURE__*/_react.default.createElement("div", {
        className: "article-title title-hover"
      }, article.title), /*#__PURE__*/_react.default.createElement("div", {
        className: "keyword-link-container bottom-border"
      }, article.categories.map(function (category, j) {
        return /*#__PURE__*/_react.default.createElement("div", {
          className: "keyword-link",
          key: "category-item-".concat(j)
        }, /*#__PURE__*/_react.default.createElement("a", {
          className: "category-item",
          href: article.link
        }, /*#__PURE__*/_react.default.createElement("span", null, category)));
      })));
    });
  };

  return /*#__PURE__*/_react.default.createElement("div", {
    className: "feed-row row__container"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "row-title"
  }, "Editorial"), _featured.default && /*#__PURE__*/_react.default.createElement("div", {
    className: "row-content"
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "editorial-article-container",
    onClick: function onClick(e) {
      return handleVisualSearch(e);
    }
  }, /*#__PURE__*/_react.default.createElement("div", {
    className: "editorial-article-title"
  }, "Find Credit Using our Visual Search Tool"), /*#__PURE__*/_react.default.createElement("img", {
    className: "editorial-article-img",
    src: _search_pad.default,
    alt: "editorial-article-img"
  })), /*#__PURE__*/_react.default.createElement("div", {
    className: "editorial-article-list"
  }, buildArticleList())));
};

var _default = EditorialRow;
exports.default = _default;
